connection.go

```diff
@@ -29,30 +29,40 @@ type StreamHandler func(stream *Stream)
 type AuthHandler func(header http.Header, slot uint8, parent uint32) bool
 
 type idleAwareFramer struct {
-	f       *spdy.Framer
-	conn    *Connection
-	expired *time.Timer
-	startCh chan struct{}
+	f              *spdy.Framer
+	conn           *Connection
+	expired        *time.Timer
```

Comment:
Possibly also hold a reference to the time channel such that it can remain nil along with expired until a non-zero timeout is issued.  I am concerned about creating this timer then stopping it, all the while it might not even be needed. Adds more logic when resetting the timer though, what do you think?

Then in the select have
```go
case <-i.expireChan:
```